onnxruntime/contrib_ops/cuda/math/bias_softmax.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "bias_softmax.h"
+#include "core/providers/common.h"
+#include "core/providers/cuda/cudnn_common.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace cuda {
+
+#define REGISTER_KERNEL_TYPED(T)                                                \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                                \
+      BiasSoftmax,                                                              \
+      kMSDomain,                                                                \
+      1,                                                                        \
+      T,                                                                        \
+      kCudaExecutionProvider,                                                   \
+      KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
+      BiasSoftmax<T>);                                                              
+
+template <typename T>
```

Comment:
Consider re-working the kernel without templates or at least the whole class being a template

Reply:
Used MLTypeCallDispatcher as suggested by Edward in the Compute(...) body.

Reply:
@yuslepukhin @edgchen1 I had to add a layer of indirection to use the dispatch mechanism. Also, does it work for more than one template argument? Is removing the top-level template supposed to have a measurable impact on binary size? I didn't observe one.

With MLTypeCallDispatcher:
```
8508    build/RelWithDebInfo_sav/libonnxruntime_common.a
135796  build/RelWithDebInfo_sav/libonnxruntime_framework.a
131360  build/RelWithDebInfo_sav/libonnxruntime_graph.a
3988    build/RelWithDebInfo_sav/libonnxruntime_mlas.a
132720  build/RelWithDebInfo_sav/libonnxruntime_optimizer.a
735372  build/RelWithDebInfo_sav/libonnxruntime_providers.a
675196  build/RelWithDebInfo_sav/libonnxruntime_providers_cuda.a
57384   build/RelWithDebInfo_sav/libonnxruntime_session.a
7628    build/RelWithDebInfo_sav/libonnxruntime_test_utils.a
22352   build/RelWithDebInfo_sav/libonnxruntime_training.a
20596   build/RelWithDebInfo_sav/libonnxruntime_training_runner.a
27704   build/RelWithDebInfo_sav/libonnxruntime_util.a
7840    build/RelWithDebInfo_sav/libonnx_test_data_proto.a
14744   build/RelWithDebInfo_sav/libonnx_test_runner_common.a
```

Without MLTypeCallDispatcher:
```
8508    build/RelWithDebInfo/libonnxruntime_common.a
135796  build/RelWithDebInfo/libonnxruntime_framework.a
131364  build/RelWithDebInfo/libonnxruntime_graph.a
3988    build/RelWithDebInfo/libonnxruntime_mlas.a
132724  build/RelWithDebInfo/libonnxruntime_optimizer.a
735372  build/RelWithDebInfo/libonnxruntime_providers.a
675320  build/RelWithDebInfo/libonnxruntime_providers_cuda.a
57384   build/RelWithDebInfo/libonnxruntime_session.a
7628    build/RelWithDebInfo/libonnxruntime_test_utils.a
22352   build/RelWithDebInfo/libonnxruntime_training.a
20596   build/RelWithDebInfo/libonnxruntime_training_runner.a
27704   build/RelWithDebInfo/libonnxruntime_util.a
7840    build/RelWithDebInfo/libonnx_test_data_proto.a
14744   build/RelWithDebInfo/libonnx_test_runner_common.a
```

Reply:
The size for build/RelWithDebInfo/libonnxruntime_providers_cuda.a is slightly smaller: 675320 (without MLTypeCallDispatcher) vs. 675196 (with).